File: Leetcode/python/sol2309.py
#https://leetcode.com/problems/greatest-english-letter-in-upper-and-lower-case/
class Solution:
    def greatestLetter(self, s: str) -> str:

        count = ["" ] *26
        for i in s:
            temp = ord(i)
            if temp < ord('a'):
                pos = temp - ord('A')
                if chr(temp) not in count[pos]:
                    count[pos]+=chr(temp)
            else:
                pos = temp - ord('a')
                if chr(temp) not in count[pos]:
                    count[pos]+=chr(temp)

        rslt = ""
        for i, j in enumerate(count[::-1]):
            if len(j) == 2:
                a = list(j)
                a.sort()
                rslt+=a[0]
                break

        return rslt

File: Leetcode/cpp/sol2309.cpp

// https://leetcode.com/problems/greatest-english-letter-in-upper-and-lower-case/description/

class Solution {
public:
    string greatestLetter(string s) {

        vector <int> lower(26, 0);
        vector <int> upper(26, 0);

        for(auto i:s){
            int val = i;
            if (val < 'a'){
                int pos = val - 'A';
                lower[pos]++;
            }else{
                int pos = val - 'a';
                upper[pos]++;
            }
        }

        for(int i = lower.size()-1; i >= 0; i--){
            if(lower[i] > 0 && upper[i] > 0){
                char character = (char)('A' + i);
                return string(1, character);
            }
        }

        return "";
    }
};